llvm/lib/Transforms/Vectorize/LoopVectorizationLegality.cpp

```diff
@@ -1514,6 +1522,27 @@ bool LoopVectorizationLegality::canVectorizeWithIfConvert() {
   return true;
 }
 
+bool LoopVectorizationLegality::canVectorizeEarlyExit() const {
```

Comment:
It feels a little odd to have both `canVectorizeEarlyExit` and `isVectorizableEarlyExitLoop` in the same class. Would it make sense to move the hints check into `isVectorizableEarlyExitLoop` to bypass legality checks? You could even get the benefit of existing code to build up a list of countable and uncountable exits that can be used later on when splitting the middle.block.

Reply:
I think you can also able enable early exit autovec with Transforms/LoopVectorize/single_early_exit.ll